src/history/test/HistoryTestsUtils.cpp

```diff
@@ -468,16 +475,36 @@ CatchupSimulation::generateRandomLedger(uint32_t version)
         {
             txs.push_back(carol.tx({payment(bob, small)}));
         }
+
+        // Add soroban transactions
+        if (protocolVersionStartsFrom(
+                lm.getLastClosedLedgerHeader().header.ledgerVersion,
+                SOROBAN_PROTOCOL_VERSION))
+        {
+            SorobanResources res;
+            res.instructions = InitialSorobanNetworkConfig::TX_MAX_INSTRUCTIONS;
```

Comment:
Is the intent here to get some of the txs applied? (I'd expect so). If so, you should use the current tx max (because initial is pretty low). You also need to set write bytes high enough.

Reply:
updated. also added a check that at least some soroban txs succeeded